fastai/callback/core.py

```diff
@@ -105,8 +105,16 @@ def before_validate(self):
 class GatherPredsCallback(Callback):
     "`Callback` that returns all predictions and targets, optionally `with_input` or `with_loss`"
     _stateattrs=('preds','targets','inputs','losses')
-    def __init__(self, with_input=False, with_loss=False, save_preds=None, save_targs=None,
-                 with_preds=True, with_targs=True, concat_dim=0, pickle_protocol=2):
+    def __init__(self,
+    with_input:bool=False, # Whether to gather and return the inputs
+    with_loss:bool=False, # Whether to gather and return the losses
+    save_preds:(str, os.PathLike)=None, # Path to save the predictions
+    save_targs:(str, os.PathLike)=None, # Path to save the targets
+    with_preds:bool=True, # Whether to gather and return the predictions
+    with_targs:bool=True, # Whether to gather and return the targets
+    concat_dim:int=0, # The dimension on which to concatenate all the tensors
+    pickle_protocol:int=2 # The pickle protocol to use for saving the predictions and targets
+    ):
```

Comment:
Suggested changes to `GatherPredsCallback.__init__`
```suggestion
    def __init__(self,
        with_input:bool=False, # Whether to return inputs
        with_loss:bool=False, # Whether to return losses
        save_preds:Path=None, # Path to save predictions
        save_targs:Path=None, # Path to save targets
        with_preds:bool=True, # Whether to return predictions
        with_targs:bool=True, # Whether to return targets
        concat_dim:int=0, # Dimension to concatenate returned tensors
        pickle_protocol:int=2 # Pickle protocol used to save predictions and targets
    ):
```